### PR TITLE
Lock frontend mission control buttons to backend response

### PR DESCRIPTION
**Pull request by eivindsjovold**

### CODE REVIEW

**Comment by phhuseboe** [comment]
Closes #386

**Comment by phhuseboe** [comment]
Closes #431

File: frontend/src/api/ApiCaller.tsx [hunk 1]
```diff
@@ -21,6 +22,19 @@ export class BackendAPICaller {
         path: string,
         body?: T
     ): Promise<{ body: T; headers: Headers }> {
+        const init: RequestInit = this.constructRequest(method, body)
```
Comment (tsundvoll) [inline frontend/src/api/ApiCaller.tsx:25]: Can `init` have a more specific name?

File: frontend/src/api/ApiCaller.tsx [hunk 2]
```diff
@@ -134,25 +140,33 @@ export class BackendAPICaller {
...
+        return this.postControlMissionRequest(path, robotId)
+    }
+
+    private async postControlMissionRequest(path: string, robotId: string): Promise<ControlMissionResponse> {
+        const url = `${config.BACKEND_URL}/${path}`
+        const body = { robotId: robotId }
+        const init: RequestInit = this.constructRequest('POST', body)
+
+        const response: Response = await fetch(url, init)
+        if (!response.ok) throw new Error(`${response.status} - ${response.statusText}`)
```
Comment (tsundvoll) [inline frontend/src/api/ApiCaller.tsx:164]: Some duplicated code with the `query` function. Can this be abstracted out? May also help with the naming of `init`?

**Review by tsundvoll — CHANGES_REQUESTED** [review]
A few comments

**Review by tsundvoll — APPROVED** [review]
Looks great!

**Pull request merged**